### PR TITLE
Port to Meson

### PR DESCRIPTION
**Pull request by Hjdskes**
This PR removes the Autotools build system for a Meson build system. For anyone watching: please test this and see if everything works as intended. It works on my system and in a Flatpak, so we should be good, but I'll leave this open to gather feedback for a few days.

### CODE REVIEW

**Comment by julianrichen** [comment]
Install & runs fine for me. The "Compile from source" README instructions need to be updated to Meson.

It's not critical but if no saved colors exists before running it throws an error on first run, program still runs:
```
$ gcolor3

** (gcolor3:10128): WARNING **: Error reading keys: Key file does not have group “Colors”
#FFFFFF
```

Probably should go into its own issue #.

**Comment by Hjdskes** [comment]
Thanks for testing! I updated the README and opened #84 for the issue.

While looking at the README, I noticed that the translation instructions say to use `intltool-update --pot` to generate an updated .pot file. Do you know if this is still the recommended way to do this, @julianrichen?

**Comment by julianrichen** [comment]
@Hjdskes I don't really know about the localization. I know Meson has the `i18n.*` functions to make translating desktop/XML files easier but I don't believe it has its own `intltool-update`. So I think it's ok?

**Comment by Hjdskes** [comment]
I'm not that familiar with localization either. I know a few years ago there was the initiative to use `AM_GNU_GETTEXT` instead, see e.g. https://blogs.gnome.org/mclasen/2016/07/21/using-modern-gettext/ and https://blogs.gnome.org/mcatanzaro/2016/07/27/on-the-killing-of-intltool/.

But I am not sure if it is still ok to use intltool for generating the .pot file. I'll ask around on IRC.

**Comment by julianrichen** [comment]
https://github.com/mesonbuild/meson/blob/master/docs/markdown/Porting-from-autotools.md#gettext

snippet:


`meson.build`:
```meson
i18n = import('i18n')

gettext_package = 'foo'
add_project_arguments('-DGETTEXT_PACKAGE=' + gettext_package, language: 'c')
subdir('po')

i18n.merge_file(
  input: 'foo.desktop.in',
  output: 'foo.desktop',
  type: 'desktop',
  po_dir: 'po',
  install: true,
  install_dir: join_paths(get_option('datadir'), 'applications')
)

i18n.merge_file(
  input: 'foo.appdata.xml.in',
  output: 'foo.appdata.xml',
  po_dir: 'po',
  install: true,
  install_dir: join_paths(get_option('datadir'), 'appdata')
)
```

`po/meson.build`:
```meson
i18n.gettext(gettext_package, preset: 'glib')
```

**Comment by Hjdskes** [comment]
Thanks, but I already have those snippets in the Meson build. Concretely, I want to know what tool to use to generate the .pot file, that translators need to make .po files. I'll ask on IRC later today or tomorrow, if you don't beat me to it ;)

**Comment by julianrichen** [comment]
Oops, my bad. Did a bit more searching: https://github.com/mesonbuild/meson/blob/master/docs/markdown/Localisation.md

So I guess?
```
$ ninja intltest-pot
```

Update .po:
```
$ ninja intltest-update-po
```

Determines name from `meson.project_name()`?

**Comment by Hjdskes** [comment]
Thanks! That works lovely :) All that seems to be left now is to go through all .po files and ensure they're correct. I'll do this tomorrow, then we can merge this!

File: org.hjdskes.gcolor3.json [hunk 1]
```diff
@@ -30,10 +30,13 @@
     "modules": [
         {
             "name": "gcolor3",
+            "buildsystem": "meson",
+	    "builddir": true,
```
Comment (jtojnar) [inline org.hjdskes.gcolor3.json]: Mixed tabs and spaces.

File: org.hjdskes.gcolor3.json [hunk 2]
```diff
@@ -30,10 +30,13 @@
...
             "sources": [
                 {
                     "type": "git",
-                    "url": "git://github.com/Hjdskes/gcolor3.git"
+                    "url": "git://github.com/Hjdskes/gcolor3.git",
+		    "branch": "meson"
```
Comment (jtojnar) [inline org.hjdskes.gcolor3.json]: Mixed tabs and spaces.

File: meson_install.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
+
+if [ -z $DESTDIR ]; then
+  PREFIX=${MESON_INSTALL_PREFIX:-/usr}
+
+  # Update icon cache
+  gtk-update-icon-cache -f -t $PREFIX/share/icons/hicolor
```
Comment (jtojnar) [inline meson_install.sh]: This will fail if datadir is changed.

File: po/meson.build [hunk 4]
```diff
@@ -0,0 +1 @@
+i18n.gettext(meson.project_name(), preset: 'glib')
```
Comment (jtojnar) [inline po/meson.build]: Cool, lowercasing fixes the metainfo localization.

File: meson_install.sh (hunk 3 above)
Comment (Hjdskes) [reply meson_install.sh]: Thanks, this should be fixed now.

**Comment by Hjdskes** [comment]
Thanks for all the help with this, @julianrichen and @jtojnar. I'm merging this now, otherwise it'll be open for another couple of months. For any fall out, please open a new issue.

**Pull request merged**